### PR TITLE
enable lint prefer_foreach

### PR DESCRIPTION
**Pull request by a14n**

### CODE REVIEW

File: packages/flutter_tools/lib/src/compile.dart [hunk 1]
```diff
@@ -131,8 +131,7 @@ class ResidentCompiler {
 
     final String inputKey = new Uuid().generateV4();
     _server.stdin.writeln('recompile $inputKey');
-    for (String invalidatedFile in invalidatedFiles)
-      _server.stdin.writeln(invalidatedFile);
+    invalidatedFiles.forEach(_server.stdin.writeln);
```
Comment (a14n) [inline packages/flutter_tools/lib/src/compile.dart]: I really don't understand why this change causes https://travis-ci.org/flutter/flutter/jobs/291503341#L3166-L3208

```
  NoSuchMethodError: The method 'call' was called on null.
  Receiver: null
  Tried calling: call("/path/to/main.dart")
  dart:core                                          List.forEach
  package:flutter_tools/src/compile.dart 134:22      ResidentCompiler.recompile
  test/compile_test.dart 178:41                      _recompile
  test/compile_test.dart 130:13                      main.<fn>.<fn>
  test/src/context.dart 102:32                       testUsingContext.<fn>.<fn>
  package:flutter_tools/src/base/context.dart 76:26  AppContext._run
  package:flutter_tools/src/base/context.dart 66:13  AppContext.runInZone.<fn>
  dart:async                                         runZoned
  package:flutter_tools/src/base/context.dart 65:12  AppContext.runInZone
  test/src/context.dart 91:32                        testUsingContext.<fn>
  ===== asynchronous gap ===========================
  dart:async                                         _Completer.completeError
  package:flutter_tools/src/compile.dart 138:3       ResidentCompiler.recompile
  ===== asynchronous gap ===========================
  dart:async                                         new Future.microtask
  package:flutter_tools/src/compile.dart 124:82      ResidentCompiler.recompile
  test/compile_test.dart 178:41                      _recompile
  ===== asynchronous gap ===========================
  dart:async                                         new Future.microtask
  test/compile_test.dart 172:36                      _recompile
  test/compile_test.dart 130:13                      main.<fn>.<fn>
  ===== asynchronous gap ===========================
  dart:async                                         _asyncThenWrapperHelper
  test/compile_test.dart 121:56                      main.<fn>.<fn>
  test/src/context.dart 102:32                       testUsingContext.<fn>.<fn>
  ===== asynchronous gap ===========================
  dart:async                                         new Future.microtask
  test/src/context.dart 91:51                        testUsingContext.<fn>.<fn>
  package:flutter_tools/src/base/context.dart 76:26  AppContext._run
  ===== asynchronous gap ===========================
  dart:async                                         new Future.microtask
  package:flutter_tools/src/base/context.dart 72:48  AppContext._run
  package:flutter_tools/src/base/context.dart 66:13  AppContext.runInZone.<fn>
  dart:async                                         runZoned
  package:flutter_tools/src/base/context.dart 65:12  AppContext.runInZone
  test/src/context.dart 91:32                        testUsingContext.<fn>
  ===== asynchronous gap ===========================
  dart:async                                         new Future.microtask
  test/src/context.dart 73:30                        testUsingContext.<fn>
```

Comment (a14n) [reply packages/flutter_tools/lib/src/compile.dart]: It seems to be an issue with mocked stdin

Comment (a14n) [reply packages/flutter_tools/lib/src/compile.dart]: fix by the last commit

**Comment by Hixie** [comment]
cc @tvolkert for the changes near the end of the PR

LGTM

**Review by tvolkert — APPROVED** [review]
LGTM

**Pull request merged**